### DISCUSSION TITLE
Leverage Circle’s compiler work on generics

### DISCUSSION THREAD

**Discussion by dvader** [carbon-language/carbon-lang#2010] [General]
Sean Baxter is doing some wicked, wicked work on generics and meta programming with his Circle C++ compiler and it works and compiles C++, it just adds a lot of nice features.  I would highly recommend Carbon takes a page from what he’s doing there. An example is his cppcon presentation https://youtu.be/15j4bkipuAg. 
1. Templates and packs
○ Templates, completed
○ Member traits
○ Circle Imperative Arguments (CIA)
○ Pack metafunctions

**Comment by jonmeow** [comment]
I'm converting this to a discussion. We use issues for things that can be addressed by a PR, and this feels more like general commentary to me (I have no real thoughts on this, we haven't really discussed metaprogramming much).

**Reply by dvader** [reply]
Yes, makes sense. Sorry I used the wrong mechanism. It is absolutely a discussion point.
